types/react-blessed/react-blessed-tests.tsx

```diff
@@ -175,7 +175,8 @@ const Box = () => <box style={{ fg: "blue" }} />;
 const BlessedBox = () => <blessed-box style={{ fg: "blue" }} />;
 const FF: React.FC<ReactBlessed.BlessedIntrinsicElements["box"]> = props => <box {...props} />;
 
-// @ts-expect-error
+// Undesired. Should error but we can only augment intrinsic elements.
```

Comment:
as in - you can't *remove* a declared existing intrinsic element? that makes sense but I wonder if react-blessed types couldnt be juggled to allow better types through `jsxImportSource` (the default case wouldn't be fixable since it relies on messing up the already declared namespaces). I also wonder, how does it even work today - is `react-blessed` "poisonous" for the project? It seems that it's messing up global~ namespace and thus it might change how a ReactDOM-based file is type-checked in the project. Did I get this right?

Reply:
The problem is on our side since we mix in DOM related things into `react`. The "intrinsics" should all be moved to `react-dom`. But I think this would be another breaking changes for projects that don't ever import from `react-dom`. 

Once that's done, `react-blessed` wouldn't need a change.

It would be really great if could separate `react` and `react-dom` properly but this is probably a lot of grunt work and I'm not even sure if it's possible while supporting all the legacy types for the factory functions.

> as in - you can't remove a declared existing intrinsic element?

I think so because declaration merging is only additive. We could potentially create an API with which you could essentially remove it but it would be specific to React and feel pretty hacky.